lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -2566,9 +2566,9 @@ HRESULT CFunctionReflection::GetDesc(D3D12_FUNCTION_DESC *pDesc) {
   pDesc->BoundResources = (UINT)m_UsedResources.size();
 
   //Unset:  UINT                    InstructionCount;            // Number of emitted instructions
-  //Unset:  UINT                    TempRegisterCount;           // Number of temporary registers used 
+  //Unset:  UINT                    TempRegisterCount;           // Number of temporary registers used
   //Unset:  UINT                    TempArrayCount;              // Number of temporary arrays used
-  //Unset:  UINT                    DefCount;                    // Number of constant defines 
+  //Unset:  UINT                    DefCount;                    // Number of constant defines
```

Comment:
For future reference, these unnecessary, albeit stylistically desirable whitespace changes really disrupt and slow the review process. I'm not opposed to them, but if you want to make them, please put them in their own change. It's really easy to review a change that only alters whitespace.

Reply:
These are made automatically by my editor to ensure _I_ don't submit unnecessary whitespace. While I understand it's hard to reject introductions of these during review it makes the code noisy to work with, both during development and as you correctly point out, review. It adds an extra barrier: I have to revert these changes, submit them separately, or turn off "trim trailing whitespace" altogether and doublecheck everything I submit.

Opened https://github.com/microsoft/DirectXShaderCompiler/pull/3272 for that, as soon as this is rebased and that's merged the changes should disappear from the diff.

On a related note it's similarly disruptive to have `.clang-format` while very few files (outside imported files from external projects/libraries) adhere to it. Only `WinAdapter` can be autoformatted without creating tons of unnecessary noise. Fixing that makes git history and blaming useless, though.

Reply:
I don't know how most people avoid these kind of incidental changes, but they do. Personally, I use emacs which lets me write what I want and doesn't do anything I don't tell it to, I find that enormously preferable. While I'm indifferent to clang formatting, I'm not a fan of clang-format. I recommend finding a way to bypass it as well.

Barring all that, we could have one enormous git blame ruining change that brings all these files into alignment with these requirements and silence this issue forever. I'm not inclined to do that myself, but I wouldn't object to it.

Reply:
> I don't know how most people avoid these kind of incidental changes, but they do.

Seems like they don't, how else did trailing whitespace end up in these files?

The point is I can easily turn it off, but it helps _me personally_ write code without ever worrying about formatting, indentation or trailing whitespace: one press of a button and it's tidy.

I don't like `clang-format`s judgement on many things either, but at least it introduces a certain form of consistency to a codebase. Still waiting for `rustfmt` to be ported to C++ :wink:.

In addition to ruining git blame, all open PRs and sidetracked branches can be thrown away, unless the user knows how to rewrite their branch efficiently in git.

Reply:
You misunderstand. I mean that people manage to avoid changes that "fix" these flaws in their PRs even though they are pervasive in the codebase.

Reply:
We both misunderstand. It is my _intention_ to leave these changes in. Usually maintainers are ever so slightly embarrassed when minor flaws like these pop up and are happy they are fixed "under the radar" rather than creating extra noise with separate commits/PRs that subsequently have to be tested (valuable CI time), approved and merged. Or left to linger around. Unless they are large enough to warrant an entry in an `ignore-revs-file`.

Reply:
I'm sorry you disapprove of how I do my job. Nevertheless, I stand by my previous statements.

Reply:
Don't get me wrong, I don't disapprove of your job at all. If anything I'm frowning upon myself for going on this tangent for this long. My sole intent is to get across _why_ I sneak such changes into a PR in the first place.

How can I best resolve this unfortunate situation? Close the whitespace PR as it's _just_ noise, and revert (as opposed to force-push off) the matching commit at the beginning of this PR?